### DISCUSSION TITLE
Fabric hangs on "Setting user: Player***"

### DISCUSSION THREAD

**Discussion by Xendergo** [FabricMC/fabric-api#1306] [Mod Dev Support] [answered]
Whenever I debug a mod, I get this warning:
```
@Mixin target net/minecraft/tag/RequiredTagList$TagWrapper is public in fabric-tag-extensions-v0.mixins.json:MixinTagImpl and should be specified in value
 com.mojang.authlib.exceptions.AuthenticationUnavailableException: null
```
and a stack trace, generally pointing to `YggdrasilSocialInteractionsService.java`, whatever that is.

Then Fabric hangs on `Setting user: Player***`

After waiting a while, I noticed the message seemed to change to `[main/ERROR] (Minecraft) Failed to verify authentication`

What's going wrong, is it just me or is it happening to everyone?

**Comment by haykam821** [comment, marked as answer]
The `@Mixin target` and `Failed to verify authentication` errors are development environment quirks and can be safely ignored.

Which Java/LWJGL versions are you using? I've had some issues with version conflicts in production where the game would hang sometime around here.

In addition, you say debugging the mod causes the issue. Does `./gradlew runClient` or a production environment from `./gradlew build` work?

**Reply by Xendergo** [reply]
runClient seems to hang too. Doing a build technically worked, but starting the game normally also made the launcher hang.

I'm using OpenJDK 11.0.9.1

**Reply by haykam821** [reply]
Which platform are you using? Also, are you using any Java GUI libraries?

**Reply by Xendergo** [reply]
I'm using AdoptOpenJDK, and I'm not using any GUI libraries

**Reply by Xendergo** [reply]
I found the issue, I was trying to create a block which stores two numbers with very large ranges in it's block state, and it seems like it was registering every possible permutation of those numbers, I had no idea it did that. I guess I'll have to use a block entity instead

**Reply by liach** [reply]
I think your computer may be just a bit slow at loading. Notice that auth error will always happen as long as you don't send a valid session token (i think) to client's command line

**Reply by YTG1234** [reply]
> Notice that auth error will always happen as long as you don't send a valid session token (i think) to client's command line

Indeed that's how it works. You can use [transfarmer's NoAuth](https://github.com/user11681/noauth) if you'd like to get rid of authentication entirely.
